api/src/main/java/org/projectnessie/tools/catalog/migration/api/CatalogMigrationResult.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.catalog.migration.api;
+
+import static org.projectnessie.tools.catalog.migration.api.CatalogMigrator.DRY_RUN_FILE;
+import static org.projectnessie.tools.catalog.migration.api.CatalogMigrator.FAILED_IDENTIFIERS_FILE;
+import static org.projectnessie.tools.catalog.migration.api.CatalogMigrator.FAILED_TO_DELETE_AT_SOURCE_FILE;
+
+import java.io.PrintWriter;
+import java.util.List;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class CatalogMigrationResult {
+
+  public abstract List<TableIdentifier> registeredTableIdentifiers();
+
+  public abstract List<TableIdentifier> failedToRegisterTableIdentifiers();
+
+  public abstract List<TableIdentifier> failedToDeleteTableIdentifiers();
+
+  public void printSummary(
```

Comment:
All these paraemters part of that params-class - same issue here - overly boilerplate and complicated.
Also - it's printing to something - that's CLI concern - not "core implemetnation"